packages/squiggle-lang/src/ast/parse.ts

```diff
@@ -94,6 +96,22 @@ export function nodeToString(
       case "Decorator":
         return sExpr([node.name, ...node.args].map(toSExpr));
       case "LetStatement":
+        if (node.typeSignature.isImplicit) {
```

Comment:
Too much copy-paste here, I recommend handling both types of statements together, and then inline-expand with:
```
[
  toSExpr(node.variable),
  node.type === "LetStatement" && !node.typeSignature.isImplicit
    ? toSExpr(node.typeSignature)
    : undefined,
  ...
]
```

(note that `sExpr` skips all undefined values)

Reply:
I tried this but it didn't work because `nodeToString` puts spaces between each entry without skipping `undefined`, so what's supposed to be (say) "Identifier UnitValue" instead gets written as "Identifier  UnitValue" (with 2 spaces). I considered adding a separate pass to remove duplicate strings but that seemed messy.

Reply:
Huh, are you sure? Maybe you tried `""` instead of `undefined`?

AFAICT `sExprToString` code filters out undefined values first.

Reply:
Turns out this does work as you suggested, I'm not sure what problem I was having with it before